src/Symfony/Component/Message/Asynchronous/Routing/ProducerForMessageResolver.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Message\Asynchronous\Routing;
+
+/**
+ * @author Samuel Roze <samuel.roze@gmail.com>
+ */
+class ProducerForMessageResolver implements ProducerForMessageResolverInterface
+{
+    /**
+     * Mapping describing which producer should be used for which message.
+     *
+     * @var array
+     */
+    private $messageToProducerMapping;
+
+    public function __construct(array $messageToProducerMapping)
+    {
+        $this->messageToProducerMapping = $messageToProducerMapping;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getProducersForMessage($message): array
```

Comment:
could this method be refactored as follows?

```php
return $this->messageToProducerMapping[get_class($message)] ?? $this->messageToProducerMapping['*'] ?? array();
```

Reply:
Good point 👍